src/wallet/wallet.h

```diff
@@ -1198,6 +1198,11 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
 
     /** Whether a given output is spendable by this wallet */
     bool OutputEligibleForSpending(const COutput& output, const CoinEligibilityFilter& eligibility_filter) const;
+
+    template<typename... Params>
```

Comment:
I also prefer this factoring. Maybe a method like `CWallet::DisplayName` is called for - the concept is used in e.g. https://github.com/bitcoin/bitcoin/blob/88a15ebc8d317a6fd4851adb344ff944d497284c/src/qt/bitcoingui.cpp#L509

Reply:
@251Labs I was hoping that there'd be an approach so that the logging user can call `LogPrintf` and not have to think about the context - the macro or some other magic would automatically add the wallet name if it's available in that context. Ultimately I think that kind of improvement would be part of a wider (structured) logging refactoring.

@Empact agree, I've factored out a `CWallet::GetDisplayName` and added calls to it in relevant parts

Reply:
Thanks @PierreRochard, yeah you're absolutely right about that. 

I was thinking about something more straightforward. 

The gist of my suggestion was to *just* extract (or remove) `WalletLogPrintf` from the wallet for the advantages that this may provide.

When extracted the signature of `WalletLogPrintf` could look something like this:

```c++
template<typename... Params> 
void WalletLogPrintf(const CWallet* wallet, const std::string& fmt, const Params&... parameters) {
    ...
};
```
and would be used like:
```c++
WalletLogPrintf(pWallet, "#reckless");
```

Imo `WalletLogPrintf` meets the recommendation: “A function that can be simply, elegantly, and efficiently implemented as a freestanding function (that is, as a nonmember function) should be implemented outside the class. ” from Stroustrup.

Anyway, this comment is just to clarify what I meant. It is not critical and it's probably best to ignore it :)